src/tests/t_kdb.py

```diff
@@ -336,6 +336,31 @@ def ldap_add(dn, objectclass, attrs=[]):
 realm.kinit('canon', password('canon'))
 realm.kinit('alias', password('canon'), ['-C'])
 
+# Test password history.
+def test_pwhist(nhist):
+    def cpw(n):
+        realm.run([kadminl, 'cpw', '-pw', str(n), princ])
+    def cpw_fail(n):
+        realm.run([kadminl, 'cpw', '-pw', str(n), princ], expected_code=1)
+    output('*** Testing password history of size %d\n' % nhist)
+    princ = 'pwhistprinc' + str(nhist)
+    pol = 'pwhistpol' + str(nhist)
+    realm.run([kadminl, 'addpol', '-history', str(nhist), pol])
+    realm.run([kadminl, 'addprinc', '-policy', pol, '-nokey', princ])
+    for i in range(nhist):
+        # Set a password, then check that all previous passwords fail.
+        cpw(i)
+        for j in range(i + 1):
+            cpw_fail(j)
+    # Set one more new password, and make sure the oldest key is
+    # rotated out.
+    cpw(nhist)
+    cpw(0)
+    cpw_fail(0)
```

Comment:
I suggested adding cpw_fail(1) before cpw(0), not after.  In the nhist == 1 case, this fails as expected because we just set a password of 1.